### PR TITLE
Initial AppleTalk fixes

### PR DESCRIPTION
**Pull request by NJRoadfan**
-Add pre-AFP 2.2 login strings.
-Add back GS/OS date preservation fix.
-Get the classic AppleTalk daemons working.
-Get the classic AppleTalk commandline tools working.

Fixes part of #1333

### CODE REVIEW

**Comment by rdmark** [comment]
@NJRoadfan Thank you for working on this. A minor housekeeping note: For PRs that target the main branch, let's not add the version number tag like [4.0]. They are only for distinguishing PRs against stable branches. Let's keep the tag soup manageable. ;)

I updated https://github.com/Netatalk/netatalk/wiki/Developer-Notes#pull-requests to make note of this.

**Comment by NJRoadfan** [comment]
I've got a2boot and timelord compiling. Right now working on getting atalkd and papd working. After that, all that is left is getting afpd back to supporting ASP transports.

**Comment by NJRoadfan** [comment]
If anyone wants to test, you need to disable ASP from libatalk for now in order to compile with AppleTalk turned on. In `libatalk/meson.build`, remove line 30 `libasp,`. ASP is only used by afpd, which has not been hacked apart to support AppleTalk yet.

File: etc/papd/uam.c [hunk 1]
```diff
@@ -60,8 +60,10 @@ struct uam_mod *uam_load(const char *path, const char *name)
 
   /* version check would go here */
 
+  /* Temporary hack below to get this to compile with 3.x, we really should be passing an AFPObj as the first parameter */
```
Comment (rdmark) [inline etc/papd/uam.c:63]: Please prefix this with a `TODO:` so that it's easy to find later.

File: libatalk/atp/atp_open.c [hunk 2]
```diff
@@ -50,9 +53,9 @@ ATP atp_open(uint8_t port, const struct at_addr *saddr)
     struct timeval	tv;
     int			pid;
 
-#ifdef DEBUG
+#ifdef EBUG
```
Comment (rdmark) [inline libatalk/atp/atp_open.c:56]: Why do you not want to use the DEBUG label here?

File: etc/meson.build [hunk 3]
```diff
@@ -5,3 +5,8 @@ subdir('afpd')
 subdir('cnid_dbd')
 subdir('netatalk')
 subdir('uams')
+
+if have_appletalk
+    subdir('atalkd')
+    subdir('papd')
+endif
```
Comment (rdmark) [inline etc/meson.build:12]: Nitpick: Let's add a final newline for consistency. :)

**Comment by rdmark** [comment]
@NJRoadfan Looks good overall. I left a few minor comments.

Can you please rebase this branch now so that it's a single commit, or a series of commits with atomic changes that compile independently? (reminder: `git rebase -i HEAD~n` where n is the number of commits that you want to squash.)

File: libatalk/atp/atp_open.c (hunk 2 above)
Comment (NJRoadfan) [reply libatalk/atp/atp_open.c:56]: Oddly, DEBUG isn't defined in the 3.x tree. It also might have been an orphan. All the other ATP source files are using EBUG in the same places for debug messages.

Comment (rdmark) [reply libatalk/atp/atp_open.c:56]: Interesting. A deep dive into git blame will probably give a hint to what happened. I agree that we should stick with the current code standard.

**Review by rdmark — APPROVED** [review]
I’m good to merge this now. Please let me know if you are working on some final touches.

**Comment by NJRoadfan** [comment]
Nothing further here.

**Pull request merged**